mango-python/bdgenomics/mango/variantViz.py

```diff
@@ -0,0 +1,442 @@
+        """
+        :param ss: SparkContext 
+        :param variantRDD: bdgenomics VariantRDD
+        :param so_term: sequence ontology term to count within genes
+        :param bins: number of bins in histograms
+        """
+
+        def compute(dd1):
+          effectList = [ effect for transcript in dd1['transcriptEffects'] for effect in transcript['effects'] ]
+          geneList = [ transcript['geneId'] for transcript in dd1['transcriptEffects'] ]
+          if len(geneList)==0:
+            geneList = list("NoGene")
+          mostSigEffect = most_sig_effect(effectList)
+          return [mostSigEffect, geneList[0]]
+
+        y= variantRDD.toDF().rdd.map(lambda w: w.asDict()).map(lambda p: p['annotation']).filter(
+          lambda x: x['attributes']['AF_NFE'] != ".").map(
+          lambda d: ( compute(d)[0],compute(d)[1])).filter(
+          lambda f: f[0] == so_term and f[1] != 'NoGene').countByValue()
+
+        self.data = y.values()
+        self.bins = bins
+        self.so_term = so_term
+
+    def plot(self, testMode = False):
+
+        if (not testMode):
+          plt.hist(self.data, self.bins)
+          plt.title("Count of " + self.so_term + " per gene")
+          plt.xlabel("Number of Variants")
+          plt.ylabel("Number of Genes")
+          plt.show()
+        return self.data
+
+class VariantsGenomicRegion(object):
+    """
+    Plot variant density across chromosome region
+    """
+    def __init__(self,ss,variantRDD, start = 0, end = 1000000000, contigName="22"):
+        """
+        :param ss: SparkContext 
+        :param variantRDD: bdgenomics variantRDD
+        :param start: genomic start position
+        :param end:  genomic stop position
+        :param contigName: contig name
+        """
+        data = variantRDD.toDF().filter((variantRDD.toDF().start > start) & (variantRDD.toDF().start < end) & (variantRDD.toDF().contigName == contigName) ).rdd.map(lambda w: w.asDict()).map(lambda x: (x['contigName'], int(math.floor(float(x['start'])/1000000)))).countByValue()
+        self.data = data
+        self.contigName=contigName
+    def plot(self):
+        x = [ z[0][1] for z in self.data.items() ]
+        data = [ (z[0][1],z[1]) for z in self.data.items() ]
+        data_sorted = sorted(data, key = lambda tup: tup[0])
+        plt.plot([ z[0] for z in data_sorted ], [ z[1] for z in data_sorted ])
+        plt.title("Variant density on chromosome " + self.contigName)
+        plt.xlabel("position (MB)")
+        plt.ylabel("Variant Count / MB")
+        plt.show()
+
+class VariantCountPerSample(object):
+    """
+    Plot Histogram of variant count per sample
+    """
+    def __init__(self,ss,genotypeRDD):
+        """
+        :param ss: SparkContext
+        :param genotypeRDD: bdgenomics genotypeRDD 
+        """
+        df =  genotypeRDD.toDF()
+        data =  df.where( (df.alleles[0] == u'ALT') | (df.alleles[1] == u'ALT')).groupBy("sampleid").count().collect()
```

Comment:
where is "sampleid" defined?

Reply:
It is a little strange to hard code it as a string if it is part of a schema

Reply:
Is there an alternative way to construct groupBy caluse  this without hardcoding it?